crates/miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -834,12 +834,12 @@ end
 export.pop_ptr_from_account_stack
     # check that the account stack always is at least of size 1, that is, it contains at least the
     # native account
-    mem_load.ACCOUNT_STACK_TOP_PTR dup push.MIN_ACCOUNT_STACK_PTR
-    gt assert.err=ERR_ACCOUNT_STACK_UNDERFLOW
+    mem_load.ACCOUNT_STACK_TOP_PTR dup
+    gt.MIN_ACCOUNT_STACK_PTR assert.err=ERR_ACCOUNT_STACK_UNDERFLOW
```

Comment:
nit (and probably not for this PR): we should be able to use the `u32assert2 u32gt` tick here to reduce cycle count.